server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java

```diff
@@ -198,44 +144,80 @@ private void handleCreate(CreationMessageEntry create, Builder responseBuilder,
             .collect(Guavate.toImmutableList());
     }
 
-    private void performCreate(CreationMessageEntry entry, Builder responseBuilder, MailboxSession session) throws MailboxException, InvalidMailboxForCreationException, MessagingException, AttachmentsNotFoundException {
+    private Builder performCreate(CreationMessageEntry entry, Builder responseBuilder, MailboxSession session) throws MailboxException, MessagingException {
         if (isAppendToMailboxWithRole(Role.OUTBOX, entry.getValue(), session)) {
-            sendMailViaOutbox(entry, responseBuilder, session);
+            return sendMailViaOutbox(entry, responseBuilder, session);
         } else if (isDraft(entry.getValue())) {
-            assertNoOutbox(entry, session);
-            saveDraft(entry, responseBuilder, session);
+            if (isTargettingAMailboxWithRole(Role.OUTBOX, entry.getValue(), session)) {
+                return invalidMailboxIds(entry, responseBuilder);
+            }
+            return saveDraft(entry, responseBuilder, session);
         } else {
             if (isAppendToMailboxWithRole(Role.DRAFTS, entry.getValue(), session)) {
-                throw new InvalidDraftKeywordsException("A draft message should be flagged as Draft");
+                return invalidDraftFlag(entry, responseBuilder);
             }
-            throw new InvalidMailboxForCreationException("The only implemented feature is sending via outbox and draft saving");
+            return invalidMailboxIds(entry, responseBuilder);
         }
     }
 
-    private void assertNoOutbox(CreationMessageEntry entry, MailboxSession session) throws MailboxException {
-        if (isTargettingAMailboxWithRole(Role.OUTBOX, entry.getValue(), session)) {
-            throw new InvalidMailboxForCreationException("Mailbox ids can combine Outbox with other mailbox");
-        }
+    private Builder invalidNotOwnedMailboxes(CreationMessageEntry create, Builder responseBuilder) {
+        return responseBuilder.notCreated(create.getCreationId(),
+            SetError.builder()
+                .type("error")
+                .properties(MessageProperty.mailboxIds)
+                .description("MailboxId invalid")
+                .build());
     }
 
-    private void assertAtLeastOneMailbox(List<MailboxId> mailboxIds) throws MailboxException {
-        if (mailboxIds.isEmpty()) {
-            throw new MessageHasNoMailboxException();
+    private Builder sendMailViaOutbox(CreationMessageEntry entry, Builder responseBuilder, MailboxSession session) throws MailboxException, MessagingException {
```

Comment:
It could be a good thing to return _SetMessagesResponse.Builder_
You have to already know the code in order to really understand what is the return.